drivers/vdpa/vfe/vfe_vdpa.c

```diff
@@ -0,0 +1,1196 @@
+	nr_vq = hw->max_queue_pairs * 2;
+	if (virtio_with_feature(hw, VIRTIO_NET_F_CTRL_VQ))
+		nr_vq += 1;
+
+	DRV_LOG(DEBUG,"virtio nr_vq is %d",nr_vq);
+
+	return nr_vq;
+}
+
+static int
+vfe_vdpa_get_queue_num(struct rte_vdpa_device *vdev, uint32_t *queue_num)
+{
+	struct vfe_vdpa_priv *priv =
+		vfe_vdpa_find_priv_resource_by_vdev(vdev);
+
+	if (priv == NULL) {
+		DRV_LOG(ERR, "Invalid vDPA device: %s.", vdev->device->name);
+		return -1;
+	}
+	// TO_DO:for net, should use max_virtqueue_pairs? may be is same
+	*queue_num = priv->hw_nr_virtqs;
+	DRV_LOG(DEBUG, "vid %d queue num is %d", priv->vid, *queue_num);
+	return 0;
+}
+
+static int
+vfe_vdpa_get_vdpa_features(struct rte_vdpa_device *vdev, uint64_t *features)
+{
+	struct vfe_vdpa_priv *priv =
+		vfe_vdpa_find_priv_resource_by_vdev(vdev);
+	struct virtio_hw *hw = NULL;
+
+	if (priv == NULL) {
+		DRV_LOG(ERR, "Invalid vDPA device: %s.", vdev->device->name);
+		return -1;
+	}
+	hw = &priv->vpdev.hw;
+	*features = hw->guest_features;
+	*features |= (1ULL << VHOST_USER_F_PROTOCOL_FEATURES);
+   
+	return 0;
+}
+
+static int
+vfe_vdpa_get_protocol_features(struct rte_vdpa_device *vdev,
+		uint64_t *features)
+{
+	struct vfe_vdpa_priv *priv =
+		vfe_vdpa_find_priv_resource_by_vdev(vdev);
+
+	if (priv == NULL) {
+		DRV_LOG(ERR, "Invalid vDPA device: %s.", vdev->device->name);
+		return -1;
+	}
+	*features = VFE_VDPA_PROTOCOL_FEATURES;
+	return 0;
+}
+
+static uint64_t
+hva_to_gpa(int vid, uint64_t hva)
+{
+	struct rte_vhost_memory *mem = NULL;
+	struct rte_vhost_mem_region *reg;
+	uint32_t i;
+	uint64_t gpa = 0;
+
+	if (rte_vhost_get_mem_table(vid, &mem) < 0)
+		goto exit;
+
+	for (i = 0; i < mem->nregions; i++) {
+		reg = &mem->regions[i];
+
+		if (hva >= reg->host_user_addr &&
+				hva < reg->host_user_addr + reg->size) {
+			gpa = hva - reg->host_user_addr + reg->guest_phys_addr;
+			break;
+		}
+	}
+
+exit:
+	free(mem);
+	return gpa;
+}
+
+static int
+vfe_vdpa_enable_vfio_intr(struct vfe_vdpa_priv *priv)
+{
+	int ret;
+	uint32_t i, nr_vring;
+	struct vfio_irq_set *irq_set;
+	int *fd_ptr;
+	struct rte_vhost_vring vring;
+	struct virtio_hw *hw = NULL;
+
+	vring.callfd = -1;
+
+	nr_vring = rte_vhost_get_vring_num(priv->vid);
+
+	irq_set = rte_zmalloc(NULL, (sizeof(struct vfio_irq_set) + sizeof(int) * (nr_vring + 1)), 0);
+	irq_set->argsz = sizeof(struct vfio_irq_set) + sizeof(int) * (nr_vring + 1);
+	irq_set->count = nr_vring + 1;
+	irq_set->flags = VFIO_IRQ_SET_DATA_EVENTFD |
+			 VFIO_IRQ_SET_ACTION_TRIGGER;
+	irq_set->index = VFIO_PCI_MSIX_IRQ_INDEX;
+	irq_set->start = 0;
+	fd_ptr = (int *)&irq_set->data;
+	fd_ptr[RTE_INTR_VEC_ZERO_OFFSET] =
+		rte_intr_fd_get(priv->pdev->intr_handle);
+
+	for (i = 0; i < nr_vring; i++) {
+		ret = rte_vhost_get_vhost_vring(priv->vid, i, &vring);
+		if (ret){
+			rte_free(irq_set);
+			DRV_LOG(ERR, "call fd get fail ret:0x%x",ret);
+			return -1;
+		}
+		fd_ptr[RTE_INTR_VEC_RXTX_OFFSET + i] = vring.callfd;
+		DRV_LOG(DEBUG, "queue %d call fd %d.", i,vring.callfd);
```

Comment:
Either use capital letter in the first character in all the logs or omit in all.
please review all DRV_LOG and have consistent notion in them.